lib/src/flutterflow_api_client.dart

```diff
@@ -265,3 +279,72 @@ Future _runFix({
     stderr.write('Error running "dart fix": $e\n');
   }
 }
+
+Future firebaseDeploy({
+  required String token,
+  required String projectId,
+  required String destinationPath,
+  String endpoint = kDefaultEndpoint,
+}) async {
+  final endpointUrl = Uri.parse(endpoint);
+  final body = jsonEncode({
+    'project': {
+      'path': 'projects/$projectId',
+    },
+    'token': token,
+  });
+  final result = await _callEndpoint(
+    client: http.Client(),
+    token: token,
+    url: Uri.https(
+        endpointUrl.host, '${endpointUrl.path}/exportFirebaseDeployCode'),
+    body: body,
+  );
+
+  // Download actual code
+  final projectZipBytes = base64Decode(result['firebase_zip']);
+  final firebaseProjectId = result['firebase_project_id'];
+  final projectFolder = ZipDecoder().decodeBytes(projectZipBytes);
+  extractArchiveToCurrentDirectory(projectFolder, destinationPath);
+  final firebaseDir = '$destinationPath/firebase';
+
+  // Install required modules for deployment.
+  await Process.run(
+    'npm',
+    ['install'],
+    workingDirectory: '$firebaseDir/functions',
+    runInShell: true,
+    stdoutEncoding: utf8,
+    stderrEncoding: utf8,
+  );
+  final directoriesResult = await Process.run(
+    'ls',
```

Comment:
I'm not sure if this will work on Windows... Let's use Dart tools for this: `Directory('...').list()` or something like that.